src/utility/cart-functions.js

```diff
@@ -0,0 +1,97 @@
+import axios from "axios";
+
+const addToCart = (cartList, dispatch, item) => {
+  const encodedToken = localStorage.getItem("token");
+
+  if (cartList.find((element) => element.id === item.id)) {
+    console.log("already present");
+  } else {
+    (async () => {
+      const response = await axios({
+        method: "post",
+        url: "/api/user/cart",
+        headers: { authorization: encodedToken },
+        data: {
+          product: item,
+        },
+      });
+      localStorage.setItem("cartList", JSON.stringify(response.data.cart));
+      dispatch({
+        type: "cartList",
+        payload: response.data.cart,
+      });
+    })();
+  }
+};
+
+const removeFromCart = (dispatch, item) => {
+  const encodedToken = localStorage.getItem("token");
+
+  (async () => {
```

Comment:
no need of async  and an iife function in #30,  is there any reason ?
instead, you can directly make removeFromCart as async